source/request_source/request_options_list_plugin_impl.cc

```diff
@@ -45,6 +45,36 @@ RequestSourcePtr OptionsListFromFileRequestSourceFactory::createRequestSourcePlu
 
 REGISTER_FACTORY(OptionsListFromFileRequestSourceFactory, RequestSourcePluginConfigFactory);
 
+std::string OptionsListFromProtoRequestSourceFactory::name() const {
+  return "nighthawk.in-line-options-list-request-source-plugin";
+}
+
+Envoy::ProtobufTypes::MessagePtr
+OptionsListFromProtoRequestSourceFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::request_source::InLinePluginConfig>();
+}
+
+RequestSourcePtr OptionsListFromProtoRequestSourceFactory::createRequestSourcePlugin(
+    const Envoy::Protobuf::Message& message, Envoy::Api::Api&,
+    Envoy::Http::RequestHeaderMapPtr header) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::request_source::InLinePluginConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  // Locking to avoid issues with multiple threads calling this at the same time and trying to set
+  // the options_list_
+  {
+    Envoy::Thread::LockGuard lock_guard(options_list_lock_);
+    // Only loading the config into memory the first time.
+    if (options_list_.options_size() == 0) {
```

Comment:
Suggestion: I think that relying on `absl::optional` here to express if options_list_ was populated slightly simplifies things when considering what happens in the edge case where the passed message actually is empty (I think right now this will repeatedly set options_list_. I see no harm in it, but it seems different from the regular flow).